### DISCUSSION TITLE
captureEnvelope and metadata

### DISCUSSION THREAD

**Discussion by farfromrefug** [getsentry/sentry-cocoa#2296] [Q&A] [answered]
I use `sentry-cocoa` in Nativescript. The way we do it is exactly the same as RN with the JS sentry creating the envelope that we send with `[PrivateSentrySDKOnly captureEnvelope:envelope]` or `[PrivateSentrySDKOnly storeEnvelope:envelope]`.

And i create the envelope from the JSON object. My issue is for those events i dont have the device context nor all the system breadcrumbs. I do see all that for a native crash (which does not go through the JS stack).

My question is when an envelope is captured is it augmented with all the native sentry-cocoa data? (like it is done on Android).

After looking at the code it does not seem to be the case. An envelope is sent right away as it is. Which means react native events (and Nativescript ones) cant be augmented with system breadcrumbs and other data.
Should i open an issue for this?

**Comment by philipphofmann** [comment, marked as answer]
> My question is when an envelope is captured is it augmented with all the native sentry-cocoa data?

No, it is not. As you already pointed out, the Cocoa SDK just sends the envelope. For adding the device context, the RN SDK uses a workaround, see https://github.com/getsentry/sentry-react-native/blob/befcc51698900298139374f7b71492b44193cfc0/ios/RNSentry.m#L164-L196

We have an issue to fix that, see https://github.com/getsentry/team-mobile/issues/11.

Maybe that also works for you, @farfromrefug.

**Reply by farfromrefug** [reply]
@philipphofmann thanks this is what i thought. I tried the RN workaround but it does not seem to work for us. The `context` in the scope is always empty.
do you have any idea why?

Also we are also missing  all native breadcrumbs. THere is no way around it? I think the java implementation parse the envelope, augments it and send it again

**Reply by philipphofmann** [reply]
> I tried the RN workaround but it does not seem to work for us.

When you don't have the `SentryCrashIntegration` running, you won't get the context. You also need to ensure that you start the Cocoa SDK, of course.

**Reply by farfromrefug** [reply]
@philipphofmann thanks i will ensure everything is good with those hints. What about the breadcrumbs?
EDIT : sorry i missed the part about the open issue ! I ll look at it to see if it resolve our case

Thanks again for everytthing

**Reply by farfromrefug** [reply]
@philipphofmann thanks a lot for your help. I managed to get everything i need! from device context to breadcrumbs thanks to your help! (which means RN can get system breadcrumbs too).

**Reply by philipphofmann** [reply]
You're welcome. It was my pleasure to help 😀
